java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessagingFactory.java

```diff
@@ -428,9 +439,14 @@ public void deregisterForConnectionError(Link link)
 	{
 		this.registeredLinks.remove(link);	
 	}
+	
+	public void scheduleOnReactorThread(final BaseHandler handler) throws IOException
+	{
+		this.getReactorScheduler().invoke(handler);
```

Comment:
invoke(...) is thread-safe.
this.workQueue is concurrent & ioSignal.sink().write(...) is thread-safe.